app/scripts/modules/kubernetes/src/validation/applicationName.validator.js

```diff
@@ -10,11 +10,16 @@ module.exports = angular
   ])
   .factory('kubernetesApplicationNameValidator', function () {
 
-    function validateSpecialCharacters(name, errors) {
+    function validateSpecialCharacters(name, warnings, errors) {
       let pattern = /^([a-zA-Z][a-zA-Z0-9]*)?$/;
       if (!pattern.test(name)) {
-        errors.push('The application name must begin with a letter and must contain only letters or digits. No ' +
-          'special characters are allowed.');
+        pattern = /^([a-zA-Z][a-zA-Z0-9-]*)?$/;
```

Comment:
The difference between these two regex patterns is quite subtle.  If we require both patterns then I suggest moving them to named constants at the top of the file to make their difference clearer.

Reply:
Yes, I considered "v1Pattern" and "v2Pattern", or would you prefer other names?

Reply:
Sure, either that or `alphanum` and `alphanumWithDash`.  Anything to indicate the difference is good.

Reply:
Done.